src/darwin/Framework/CHIP/CHIPDeviceController.mm

```diff
@@ -180,17 +180,16 @@ - (NSNumber *)getControllerNodeId
 
 - (NSNumber *)_getControllerNodeId
 {
-    uint16_t idStringLen = 32;
-    char deviceIdString[idStringLen];
+    uint16_t deviceIdLength = sizeof(_localDeviceId);
     if (CHIP_NO_ERROR
-        != _persistentStorageDelegateBridge->SyncGetKeyValue(CHIP_COMMISSIONER_DEVICE_ID_KEY, deviceIdString, idStringLen)) {
-        _localDeviceId = arc4random();
```

Comment:
Unintentional removal ?

Reply:
Yes, added back. Thank you for noticing.